libs/community/langchain_community/llms/mlrun.py

```diff
@@ -0,0 +1,170 @@
+from collections.abc import AsyncIterator, Iterator
+from typing import Any, List, Optional, Union
+
+import mlrun
+from langchain_core.language_models.base import LanguageModelInput
+from langchain_core.language_models.llms import LLM
+from langchain_core.pydantic_v1 import Field
+from langchain_core.runnables import RunnableConfig
+
+
+class Mlrun(LLM):
+    """A class for interacting with Langchain, while using models served by mlrun."""
+
+    llm: Any = Field(default=None)
+    name: str = Field(default=None)
+
+    def __init__(
+        self,
+        nuclio_function: Any,
+        name: str,
+        **kwargs: dict,
+    ):
+        super().__init__(**kwargs)
+        self.llm = nuclio_function
+        self.name = name
+
+    @property
+    def _llm_type(self) -> str:
+        """Get the type of language model used by this chat model.
+        Used for logging purposes only."""
+        return "Mlrun"
+
+    def _call(
+        self,
+        prompt: str,
+        **generate_kwargs: dict,
+    ) -> str:
+        pass
+
+    def invoke(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> str:
+        """
+        call the llm invoke function, routed through mlrun model server.
+        """
+        # Convert the input to the correct format
+        input = self._convert_input(input)
+        # Check if the model server is a mock server or a real server
+        # and call the correct function
+        if isinstance(self.llm, mlrun.serving.server.GraphServer):
+            response = self.llm.test(
+                path=f"/v2/models/{self.name}/predict",
+                body={
+                    "inputs": [input],
+                    "stop": stop,
+                    "config": config,
+                    "usage": "invoke",
+                    **kwargs,
+                },
+            )
+            return response if type(response) is str else response["outputs"]
+        else:
+            response = self.llm.invoke(
+                path=f"/v2/models/{self.name}/predict",
+                body={
+                    "inputs": [input],
+                    "stop": stop,
+                    "config": config,
+                    "usage": "invoke",
+                    **kwargs,
+                },
+            )
+            return response if type(response) is str else response["outputs"]
+
+    async def ainvoke(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> str:
+        """
+        call the llm ainvoke function, routed through mlrun model server.
+        """
+        raise NotImplementedError("ainvoke is not supported by mlrun yet.")
```

Comment:
Please wrap methods or function names in ``
```suggestion
        raise NotImplementedError("`ainvoke` is not supported by mlrun yet.")
```